AzFilesHybrid/AzFilesHybrid.psm1

```diff
@@ -4295,41 +4297,27 @@ function Update-AzStorageAccountAuthForAES256 {
             -ResourceGroupName $ResourceGroupName -StorageAccountName $StorageAccountName -ErrorAction Stop
         $domain = $activeDirectoryProperties.DomainName
 
-        if ($adObject.ObjectClass -ine "computer") {
-            $message = "Removing object '$($adObject.DistinguishedName)' of type '$adObject.ObjectClass' from domain '$domain'." `
-                + " AES256 is only supported for computer objects."
-            Write-Verbose -Message $message
-
-            Remove-ADObject -Identity $adObject.DistinguishedName -Server $domain -Confirm:$false -ErrorAction Stop
-
-            $organizationalUnitDistinguishedName = $adObject.DistinguishedName.Substring($adObject.DistinguishedName.IndexOf(',') + 1)
-            $samAccountName = $adObject.SamAccountName.TrimEnd("$")
+        Set-StorageAccountDomainProperties `
+            -ADObjectName $adObjectName `
+            -ResourceGroupName $ResourceGroupName `
+            -StorageAccountName $StorageAccountName `
+            -Domain $domain `
+            -Force

```

Comment:
Move Set-StorageAccountDomainProperties after Set-ADUser/Set-ADComputer. Should update storage account only after the AD object is properly configured.

Reply:
updated